File: ruby/max_product.rb
def max_product(array)
	max, second_max, third_max = 0

	array.each do |number|
		if number >= max
			third_max = second_max
			second_max = max
			max = number
		elsif number >= second_max
			third_max = second_max
			second_max = number
		elsif number >= third_max
			third_max = number
		end
	end
	return max * second_max * third_max
end

puts max_product([100, 2, 3 ,4 ,20, 20, 400])

File: python/max_product.py
def max_product(array):
	maximum, second_max, third_max = [0, 0, 0]

	for number in array:
		if number >= maximum:
			third_max = second_max
			second_max = maximum
			maximum = number
		elif number >= second_max:
			third_max = second_max
			second_max = number
		elif number >= third_max:
			third_max = number

	return maximum * second_max * third_max

print(max_product([100, 2, 3 ,4 ,20, 20, 400]))
